src/xdp/trn_agent_xdp.c

```diff
@@ -120,6 +124,16 @@ static __inline int trn_encapsulate(struct transit_packet *pkt,
 		d_addr = dst_r_ep->ip;
 		d_mac = dst_r_ep->mac;
 
+		__u32 inhash = jhash_2words(in_src_ip, in_dst_ip, INIT_JHASH_SEED);
+		s_port = SPORT_MIN + (inhash % SPORT_BASE);
+		if (s_port < SPORT_MIN || s_port > SPORT_MAX) {
+			bpf_debug("[Agent:%ld.0x%x] DROP (BUG): s_port[%d] "
+			 		" is outside of port range!\n",
+					pkt->agent_ep_tunid, bpf_ntohl(pkt->agent_ep_ipv4),
+			  		s_port);
+			return XDP_DROP;
+		}
+
```

Comment:
Looks like this could be causing a failure in the HTTP/TCP tests. Can you rerun the tests to see if it's transient?

Reply:
I reran the e2efunctest and there's no complaints. Is this the one you're talking about?

Reply:
No, under the "Checks" tab of the PR, you can rerun all tests on the CI

Reply:
Just reran it, seems all passed.